torch/csrc/jit/codegen/cuda/lower_index_hoist.cpp

```diff
@@ -173,13 +173,22 @@ std::pair<Val*, bool> CommonIndexMap::insert(
     const std::vector<kir::ForLoop*>& loops,
     Val* index) {
   if (index->definition() == nullptr) {
-    // Only expression is eligible to hoist
+    // Only defined val is eligible to hoist
     return {index, false};
   }
 
   const CommonIndexKey key(
       indexed_consumer_id, consumer_td, ref_td, ref_index_map, loops);
 
+  // Hoisting is not possible if any of used loops is grouped.
```

Comment:
I don't think this would be an issue. In matmul patterns the values to reduce would be in registers, so no need to hoist the tensor index at this point, and most use cases should be cascading reductions, i.e. rfactors, so predicate would be removed.